src/main/java/uk/gov/hmcts/ccd/data/draft/DefaultDraftGateway.java

```diff
@@ -0,0 +1,179 @@
+            appInsights.trackDependency(DRAFT_STORE, "Update", duration.toMillis(), true);
+            return aDraftResponse()
+                .withId(draftId)
+                .build();
+        } catch (HttpClientErrorException e) {
+            LOG.warn("Error while updating draftId={}", draftId, e);
+            if (e.getRawStatusCode() == RESOURCE_NOT_FOUND) {
+                throw new ResourceNotFoundException(String.format(RESOURCE_NOT_FOUND_MSG, draftId));
+            }
+        } catch (Exception e) {
+            LOG.warn("Error while updating draftId={}", draftId, e);
+            throw new ServiceException(DRAFT_STORE_DOWN_ERR_MESSAGE, e);
+        }
+        return null;
+    }
+
+    @Override
+    public DraftResponse get(final String draftId) {
+        try {
+            HttpHeaders headers = securityUtils.authorizationHeaders();
+            headers.add(DRAFT_ENCRYPTION_KEY_HEADER, applicationParams.getDraftEncryptionKey());
+            final HttpEntity requestEntity = new HttpEntity(headers);
+            final Instant start = Instant.now();
+            Draft draft = restTemplate.exchange(applicationParams.draftURL(draftId), HttpMethod.GET, requestEntity, Draft.class).getBody();
+            final Duration duration = Duration.between(start, Instant.now());
+            appInsights.trackDependency(DRAFT_STORE, "Get", duration.toMillis(), true);
+            return assembleDraft(draft);
+        } catch (HttpClientErrorException e) {
+            LOG.warn("Error while getting draftId=" + draftId, e);
+            if (e.getRawStatusCode() == RESOURCE_NOT_FOUND) {
+                throw new ResourceNotFoundException(String.format(RESOURCE_NOT_FOUND_MSG, draftId));
+            }
+        } catch (Exception e) {
+            LOG.warn("Error while getting draftId=" + draftId, e);
+            throw new ServiceException(DRAFT_STORE_DOWN_ERR_MESSAGE, e);
+        }
+        return null;
+    }
+
+    @Override
+    public List<DraftResponse> getAll() {
+        try {
+            HttpHeaders headers = securityUtils.authorizationHeaders();
+            headers.add(DRAFT_ENCRYPTION_KEY_HEADER, applicationParams.getDraftEncryptionKey());
+            final HttpEntity requestEntity = new HttpEntity(headers);
+            final Instant start = Instant.now();
+            DraftList getDrafts = restTemplate.exchange(getUriWithQueryParams(), HttpMethod.GET, requestEntity, DraftList.class).getBody();
+            final Duration duration = Duration.between(start, Instant.now());
+            appInsights.trackDependency(DRAFT_STORE, "GetAll", duration.toMillis(), true);
+            return getDrafts.getData()
+                .stream()
+                .map(this::assembleDraft)
+                .collect(Collectors.toList());
+        } catch (Exception e) {
+            LOG.warn("Error while getting drafts", e);
+            throw new DraftAccessException(DRAFT_STORE_DOWN_ERR_MESSAGE, e);
```

Comment:
same as above, this could be a bit misleading.

Reply:
This is as expected. Per design a generic message is supposed to be returned regardless of the reason.